### PR TITLE
Create Command_requesthelp.java

### PR DESCRIPTION
**Pull request by Scuph**
Since a lot of admins can't see someone asking for help because our constant chat flood due to our player amount, I made this.

### CODE REVIEW

**Comment by AlexFreedomMod** [comment]
How do the staff know when they need help??

**Comment by AlexFreedomMod** [comment]
You should make it so it sends the admins a message.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.ChatColor;
+
+@CommandPermissions(level = AdminLevel.ALL, source = SourceType.BOTH)
+@CommandParameters( description = "Request help", usage = "/<command>", aliases = "ineedhelp") {
+
```
Comment (AlexFreedomMod) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: @Scuph, There shouldn't be a space between the ( and the word description

**Comment by Scuph** [comment]
@AlexFreedomMod How do you send only admins a message?

**Comment by AlexFreedomMod** [comment]
......
If (TFM_AdminList.isSuperAdmin())

**Comment by SlicedCaek** [comment]
Something like
```If (TFM_AdminList.isSuperAdmin())
{
    Send something
}
```
After {, you should add something like AdminChat or similar to that...

**Comment by AlexFreedomMod** [comment]
Yep!, @SlicedCaek is right

**Comment by AlexFreedomMod** [comment]
[code]
test

**Comment by Scuph** [comment]
I have added everything.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.ChatColor;
+
+@CommandPermissions(level = AdminLevel.ALL, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Request help", usage = "/<command>", aliases = "ineedhelp") {
+
+    @Override
+    public boolean run(CommandSender sender, Command cmd, String commandLabel, String[] args) {
+        TFM_Util.adminAction(sender.getName() + "Requesting help");
+        if (TFM_AdminList.isSuperAdmin())
```
Comment (SlicedCaek) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: it should be
``` java
if (TFM_AdminList.isSuperAdmin(player))
```
Correct me if i am wrong

Comment (SlicedCaek) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: It should  be something like
``` java
if (TFM_AdminList.isSuperAdmin(player))
```
Correct me if I'm wrong

**Comment by JeromSar** [comment]
@Scuph This doesn't compile, let alone be free. Did you read the contributing guidelines? They exist for a reason, you know.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    player.sendMessage(sender.getName() + " has requested help.");
```
Comment (AlexFreedomMod) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: Who's player?? My netbeans says "???"

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java (hunk 2 above)
Comment (AlexFreedomMod) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: It's right but, who's player nothing states what player is.

**Comment by Scuph** [comment]
@JeromSar fixed.

**Comment by Wild1145** [comment]
Formatting's shot to pieces as well...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+@CommandPermissions(level = AdminLevel.ALL, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Request help", usage = "/<command>", aliases = "ineedhelp") {
+
+    @Override
+    public boolean run(CommandSender sender, Command cmd, String commandLabel, String[] args) {
+        TFM_Util.adminAction(sender.getName() + "Requesting help");
+        if (TFM_AdminList.isSuperAdmin(player))
+{
+    player.sendMessage(sender.getName() + ChatColor.RED + " has requested help.");
+}
```
Comment (AlexFreedomMod) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: Umm. Net beans doesn't know who player is....

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import me.StevenLawson.TotalFreedomMod.TFM_PlayerData;
+import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.ChatColor;
+
+@CommandPermissions(level = AdminLevel.ALL, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Request help", usage = "/<command>", aliases = "ineedhelp") {
```
Comment (AlexFreedomMod) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: This is not a class.

**Comment by AlexFreedomMod** [comment]
I'll fix the pull request by making another one

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Override
+    public boolean run(CommandSender sender, Command cmd, String commandLabel, String[] args) {
+        TFM_Util.adminAction(sender.getName() + "Requesting help");
+        if (TFM_AdminList.isSuperAdmin(player))
+{
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_requesthelp.java]: Formatting

**Comment by JeromSar** [comment]
@Scuph There's still lots to consider here:

First of all, is there an issue linking to this PR. Do really want this feature?

Secondly, not only does this change not work, it also doesn't even compile. The entire structure of the command is off and its rendered completely dysfunctional.

I suggest you read up on beginner java and Bukkit plugin tutorials before contributing to TFM.

**Pull request closed without merging**